### PR TITLE
[enhancement] Allow users to specify Django version #313

### LINKED ISSUE #313: [enhancement] Allow users to specify Django version
Add a variable to specify which version of Django should be installed. 
This task should be run after everything is installed by pip to avoid accidental upgrades by any other package. 

I think we should default it to `~=3.2.0`, since it is an LTS release of Django (open for discussion).

### PR DESCRIPTION
**Pull request by yashikajotwani12**
Fixes #313

### CODE REVIEW

File: tasks/pip.yml [hunk 1]
```diff
@@ -328,3 +328,18 @@
   register: result
   until: result is success
   notify: reload supervisor
+
+- name: Install django
+  pip:
+    name: "django~=3.2.0"
```
Comment (pandafy) [inline tasks/pip.yml:334]: @yashikajotwani12  we need to add a variable that contains `~=3.2.0`. 

```suggestion
    name: "django{{ openwisp2_django_version}}"
```

This variable should be added in https://github.com/openwisp/ansible-openwisp2/blob/master/defaults/main.yml with the default value of `~=3.2.0`. 

Please also update the ["Role Variables" section of the README](https://github.com/openwisp/ansible-openwisp2#role-variables) to add this new variable.

**Review by pandafy — CHANGES_REQUESTED** [review]
Good start @yashikajotwani12! See my comment below.

File: tasks/pip.yml (hunk 1 above)
Comment (yashikajotwani12) [reply tasks/pip.yml:334]: Yeah! I will update these changes soon, Thanks

File: defaults/main.yml [hunk 2]
```diff
@@ -13,6 +13,7 @@ openwisp2_network_topology_version: "0.5.1"
 openwisp2_firmware_upgrader_version: "0.1"
 openwisp2_monitoring_version: "0.1"
 openwisp2_radius_version: "0.2.1"
+openwisp2_django_version: "3.2.0"
```
Comment (pandafy) [inline defaults/main.yml:16]:
```suggestion
openwisp2_django_version: "~=3.2.0"
```
It won't work in the current state.

File: tasks/pip.yml [hunk 3]
```diff
@@ -328,3 +328,18 @@
   register: result
   until: result is success
   notify: reload supervisor
+
+- name: Install django
+  pip:
+    name: "django{{ openwisp2_django_version}}"
```
Comment (pandafy) [inline tasks/pip.yml:334]: In the current state, this line will evaluate to `django3.2.0` which is wrong.

**Review by pandafy — CHANGES_REQUESTED** [review]

File: tasks/pip.yml (hunk 3 above)
Comment (yashikajotwani12) [reply tasks/pip.yml:334]: Okay, I understood. I am improving this. Thanks

Comment (nemesisdesign) [reply tasks/pip.yml:334]:
```suggestion
    name: "django~={{ openwisp2_django_version}}"
```

File: defaults/main.yml [hunk 4]
```diff
@@ -13,6 +13,7 @@ openwisp2_network_topology_version: "0.5.1"
 openwisp2_firmware_upgrader_version: "0.1"
 openwisp2_monitoring_version: "0.1"
 openwisp2_radius_version: "0.2.1"
+openwisp2_django_version: "~=3.2.0"
```
Comment (nemesisdesign) [inline defaults/main.yml:16]: This cant' be so inconsistent with the rest of the code, moreover, the README doesn't reflect this, too many wrongs here.

```suggestion
openwisp2_django_version: "3.2.0"
```

**Review by nemesisdesign — CHANGES_REQUESTED** [review]

**Review by nemesisdesign — APPROVED** [review]

**Pull request merged**